app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt

```diff
@@ -44,18 +48,23 @@ class WallpaperSettingsFragment : Fragment() {
             setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
             setContent {
                 FirefoxTheme {
-                    var currentWallpaper by remember { mutableStateOf(wallpaperManager.currentWallpaper) }
+                    val wallpapers = appStore.observeAsComposableState { state ->
+                        state.wallpaperState.availableWallpapers
+                    }.value ?: listOf()
+                    var currentWallpaper = appStore.observeAsComposableState { state ->
+                        state.wallpaperState.currentWallpaper
+                    }.value ?: Wallpaper.Default
                     var wallpapersSwitchedByLogo by remember { mutableStateOf(settings.wallpapersSwitchedByLogoTap) }
                     WallpaperSettings(
-                        wallpapers = wallpaperManager.wallpapers,
-                        defaultWallpaper = WallpaperManager.defaultWallpaper,
+                        wallpapers = wallpapers,
+                        defaultWallpaper = Wallpaper.Default,
                         loadWallpaperResource = { wallpaper ->
-                            with(wallpaperManager) { wallpaper.load(context) }
+                            wallpaperUseCases.loadBitmap(wallpaper)
                         },
                         selectedWallpaper = currentWallpaper,
                         onSelectWallpaper = { selectedWallpaper: Wallpaper ->
                             currentWallpaper = selectedWallpaper
```

Comment:
the lambdas for `loadWallpaperResource`, `onSelectWallpaper`, and `onTapLogoSwitchCheckedChange` are pretty substantial. Would it make sense to abstract this out into an interactor/controller? It would keep the instantiation clean and allow for the business logic to be testable.

Reply:
`loadWallpaperResource` doesn't seem like it would benefit hugely, as it would just change from `wallpaperUseCases.loadBitmap(wallpaper) to `wallpaperSettingsInteractor.loadBitmap(wallpaper)`.

For the others, they certainly do look out of place in terms of size. It looks like they're just recording metrics though, and could otherwise be a single line as well. Would the metrics be an appropriate thing to just move into the use case, or is an interactor a better place? I'm just a little hesitant to add another layer of abstraction if we can find a way to avoid it.

`onTapLogoSwitchCheckedChange` should also be removed as part of https://github.com/mozilla-mobile/fenix/issues/25985 so we can potentially ignore that for now as well. Come to think of it, should the metric be removed directly from `metrics.yaml` as part of that work or is there a specific process for removing telemetry? Unrelated but would be good to know for later

Reply:
Ah, it looks like there was actually some additional logic in `onSelectWallpaper` that can be removed entirely now that the wallpaper data is being observed from the store. I'll update that accordingly and move the metrics in the UseCase for now. If that's not a good place for the metrics to live we can revisit, but it should at least address the concern of high lambda complexity

Reply:
I feel that anything not related to the UI should live within an interactor/controller, to keep the instantiation here clean and keep the UI as dumb as possible. It's entirely possible that, long term, we move away from Fragments and go 100% Compose navigation, and it'll be easier at that time just dropping-in the interactor rather than having to refactor the fragment AND the business logic. Additionally, the use case operations could be abstracted there as well. 

As for metrics, I'd make a separate PR for removing them from `metrics.yaml` and doing whatever other clean up and then tagging-in Roger for the blessing, since he's a data steward. Anecdotally, I didn't have to do anything additional admin for my search term tab groups removal.

Reply:
> and it'll be easier at that time just dropping-in the interactor rather than having to refactor the fragment AND the business logic.

My intention was to have the UseCase take this role. The latest changes hopefully make that a little more clear. I personally would prefer not to wrap the UseCase in an interactor/controller, as I worry that they would act essentially as passthrough objects and I think that can make our codebase more difficult to navigate. I may not be fully understanding your suggestion or our patterns though. My concern is basically that this:

```
loadWallpaperResource = { wallpaperUseCases.loadBitmap(it) }
```

seems more straightforward than this:
```
loadWallpaperResource = { interactor.loadBitmap(it) }

interface WallpaperSettingsInteractor {
    fun loadWallpaperBitmap(wallpaper: Wallpaper)
}

class SessionControlInteractor(val wallpaperSettingsController: WallpaperSettingsController) : WallpaperSettingsInteractor {
    override fun loadWallpaperBitmap(wallpaper: Wallpaper) {
        wallpaperSettingsController.loadWallpaperBitmap(wallpaper)
    }
}

class WallpaperSettingsController(val useCases: WallpaperUseCases) {
    fun loadWallpaperBitmap(wallpaper: Wallpaper) {
        useCases.loadBitmap(it)
    }
}
```

> As for metrics, I'd make a separate PR for removing them from metrics.yaml and doing whatever other clean up and then tagging-in Roger for the blessing, since he's a data steward. Anecdotally, I didn't have to do anything additional admin for my search term tab groups removal.

Great, thanks!

Reply:
I think the main thing I was looking for was the abstraction, and the way you wrote the use cases seems to do that well and could be a 2.0, cleaner, more direct version of the interactor/controller pattern. You also have tests written for them, so ✅